rasa/core/policies/ensemble.py

```diff
@@ -91,590 +218,316 @@ def _check_for_important_policies(self) -> None:
-                    (
-                        state_featurizer_func,
-                        state_featurizer_config,
-                    ) = cls.get_state_featurizer_from_dict(featurizer_config)
-
-                    # override featurizer's state_featurizer
-                    # with real state_featurizer class
-                    featurizer_config["state_featurizer"] = state_featurizer_func(
-                        **state_featurizer_config
-                    )
-
-                # override policy's featurizer with real featurizer class
-                policy["featurizer"] = featurizer_func(**featurizer_config)
-
-            policy_name = policy.pop("name")
-            try:
-                constr_func = registry.policy_from_module_path(policy_name)
-                try:
-                    policy_object = constr_func(**policy)
-                except TypeError as e:
-                    raise Exception(f"Could not initialize {policy_name}. {e}") from e
-                parsed_policies.append(policy_object)
-            except (ImportError, AttributeError) as e:
-                raise InvalidPolicyConfig(
-                    f"Module for policy '{policy_name}' could not "
-                    f"be loaded. Please make sure the "
-                    f"name is a valid policy."
-                ) from e
-
-        return parsed_policies
-
-    @classmethod
-    def get_featurizer_from_dict(cls, policy: Dict[Text, Any]) -> Tuple[Any, Any]:
-        """Gets the featurizer initializer and its arguments from a policy config."""
-        # policy can have only 1 featurizer
-        if len(policy["featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every policy can only have 1 featurizer "
-                f"but '{policy.get('name')}' "
-                f"uses {len(policy['featurizer'])} featurizers."
-            )
-        featurizer_config = policy["featurizer"][0]
-        featurizer_name = featurizer_config.pop("name")
-        featurizer_func = registry.featurizer_from_module_path(featurizer_name)
-
-        return featurizer_func, featurizer_config
-
-    @classmethod
-    def get_state_featurizer_from_dict(
-        cls, featurizer_config: Dict[Text, Any]
-    ) -> Tuple[Callable, Dict[Text, Any]]:
-        # featurizer can have only 1 state featurizer
-        if len(featurizer_config["state_featurizer"]) > 1:
-            raise InvalidPolicyConfig(
-                f"Every featurizer can only have 1 state "
-                f"featurizer but one of the featurizers uses "
-                f"{len(featurizer_config['state_featurizer'])}."
-            )
-        state_featurizer_config = featurizer_config["state_featurizer"][0]
-        state_featurizer_name = state_featurizer_config.pop("name")
-        state_featurizer_func = registry.state_featurizer_from_module_path(
-            state_featurizer_name
-        )
+    def validate(self, ensemble: List[Policy], domain: Domain) -> None:
```

Comment:
can we dump the whole method then? I think it's good to add later but as far as I can see it's not a check we have currently and gives just one more thing to take care of for the architecture revamp otherwise